src/app/views/provisioning/details.controller.coffee

```diff
@@ -1,34 +1,84 @@
-@App.controller('ProvisioningDetailsCtrl', ($state, $stateParams, MnoeProvisioning, schemaForm) ->
+@App.controller('ProvisioningDetailsCtrl', ($q, $state, $stateParams, MnoeProvisioning, MnoeOrganizations, schemaForm, PRICING_TYPES, EDIT_ACTIONS) ->
   vm = this
 
   vm.form = [ "*" ]
-
   vm.subscription = MnoeProvisioning.getSubscription()
 
-  # Happen when the user reload the browser during the provisioning
+
+  urlParams =
+    orgId: $stateParams.orgId,
+    id: $stateParams.id,
+    nid: $stateParams.nid,
+    editAction: $stateParams.editAction
+
+  getCustomSchema = (product) ->
+    # The schema is contained in field vm.product.custom_schema
+    #
+    # jsonref is used to resolve $ref references
+    # jsonref is not cyclic at this stage hence the need to make a
+    # reasonable number of passes (2 below + 1 in the sf-schema directive)
+    # to resolve cyclic references
+    schemaForm.jsonref(JSON.parse(product.custom_schema))
+      .then((schema) -> schemaForm.jsonref(schema))
+      .then((schema) -> schemaForm.jsonref(schema))
+      .then((schema) ->
+        vm.schema = schema
+      )
+
   if _.isEmpty(vm.subscription)
-    # Redirect the user to the first provisioning screen
-    $state.go('dashboard.provisioning.order', {orgId: $stateParams.orgId, id: $stateParams.id, nid: $stateParams.nid}, {reload: true})
+    vm.isLoading = true
```

Comment:
Because when editing a subscription we jump directly to this page, we need to sometimes fetch the subscription/product here.